panther_battery/src/battery_node.cpp

```diff
@@ -26,31 +28,52 @@
 #include <panther_battery/battery.hpp>
 #include <panther_battery/battery_publisher.hpp>
 #include <panther_battery/dual_battery_publisher.hpp>
+#include <panther_battery/roboteq_battery.hpp>
 #include <panther_battery/single_battery_publisher.hpp>
 
 namespace panther_battery
 {
-using std::placeholders::_1;
 
-ADCNode::ADCNode(const std::string & node_name, const rclcpp::NodeOptions & options)
+BatteryNode::BatteryNode(const std::string & node_name, const rclcpp::NodeOptions & options)
 : Node(node_name, options)
 {
-  this->declare_parameter<std::string>("adc0_device", "/sys/bus/iio/devices/iio:device0");
-  this->declare_parameter<std::string>("adc1_device", "/sys/bus/iio/devices/iio:device1");
+  this->declare_parameter<float>("panther_version", 1.2);
   this->declare_parameter<int>("ma_window_len/voltage", 10);
   this->declare_parameter<int>("ma_window_len/current", 10);
-  this->declare_parameter<int>("ma_window_len/temp", 10);
-  this->declare_parameter<int>("ma_window_len/charge", 10);
 
   // running at 10 Hz
   battery_pub_timer_ = this->create_wall_timer(
-    std::chrono::milliseconds(100), std::bind(&ADCNode::BatteryPubTimerCB, this));
+    std::chrono::milliseconds(100), std::bind(&BatteryNode::BatteryPubTimerCB, this));
 
   RCLCPP_INFO(this->get_logger(), "Node started");
 }
 
-void ADCNode::Initialize()
+void BatteryNode::Initialize()
 {
+  auto panther_version = this->get_parameter("panther_version").as_double();
+  if (panther_version < 1.2 - std::numeric_limits<float>::epsilon()) {
```

Comment:
```suggestion
```suggestion
  const float panther_version = this->get_parameter("panther_version").as_double();
  if (panther_version < 1.2f - std::numeric_limits<float>::epsilon()) {
```
This way, the double will be automatically cast to float and the computation will not make any pointless recasting.

One question though, why do you subtract the epsilon?

Reply:
To ensure condition is not fulfilled when panthet_version equals precisely 1.2

Reply:
Adding brackets surrounding the subtraction will make the order more clear in this case